### PR TITLE
[WIP POC] fix timer problem

### PR DESCRIPTION
**Pull request by jsedgwick**
turns out timers are not an ongoing thing, you need to reset them after they've fired

also now you now what WIP looks like from me i.e. total shitshow e.g. `TestFuckMe` and associated clowniness. Obvi needs more than a little cleanup, but without that timer Reset() call this fails. Passes under `-race` and so on with the timer reset, fails (waiting to fill up buffer, presumably) without

This does not address the stat-splitting across buffers issue

### CODE REVIEW

**Comment by jsedgwick** [comment]
PTAL @rodaine @btc @junr03

File: tcp_sink.go [hunk 1]
```diff
@@ -112,19 +135,20 @@ func (s *tcpStatsdSink) run() {
...
 				time.Sleep(3 * time.Second)
 				continue
 			}
-			s.conn = conn
+			s.c = c
 		}
 
 		select {
 		case <-t.C:
+			t.Reset(time.Second)
```
[the remark this answers is not in the document]
Comment (btc) [reply tcp_sink.go]: damn. it was supposed to be a ticker. not a timer but a ticker. that’s what we want

**Comment by btc** [comment]
the timer should be a ticker

**Comment by jsedgwick** [comment]
ok we'll fix this proper and do all the testing things next week

**Comment by jsedgwick** [comment]
gonna just let this sit here for now, timeboxing :D

**Pull request closed without merging**